go/database/mpt/nodes_test.go

```diff
@@ -6822,6 +6822,86 @@ func TestVisitPathToStorage_CanReachTerminalNodes(t *testing.T) {
 	}
 }
 
+func TestVisitPathToStorage_EmbeddedNode_Flag_Tracked(t *testing.T) {
+	address := common.Address{1}
+	key := common.Key{2}
+	key2 := common.Key{3}
+	var value common.Value
+	value[20] = 0x02
+	value[21] = 0x04
+
+	ctrl := gomock.NewController(t)
+	ctxt := newNodeContext(t, ctrl)
+
+	tests := map[string]struct {
+		desc     NodeDesc
+		keyFound bool
+	}{
+		"matching embedded": {
+			desc: &Extension{
+				path: AddressToNibblePath(address, ctxt)[0:30],
+				next: &Account{address: address, pathLength: 34, info: AccountInfo{Nonce: common.Nonce{0x01}},
+					storage: &Extension{
+						path:         KeyToNibblePath(key, ctxt)[0:40],
+						nextEmbedded: true,
+						next:         &Value{key: key, length: 24, value: value},
+					}}},
+			keyFound: true,
+		},
+		"mismatch embedded": {
+			desc: &Extension{
+				path: AddressToNibblePath(address, ctxt)[0:30],
+				next: &Account{address: address, pathLength: 34, info: AccountInfo{Nonce: common.Nonce{0x01}},
+					storage: &Extension{
+						path:         KeyToNibblePath(key, ctxt)[0:40],
+						nextEmbedded: true,
+						next:         &Value{key: key2, length: 24, value: value},
+					}}},
```

Comment:
These two setups are nearly identical - the only difference is the key used in the Value node, right?

How about replacing the `desc` field in the test specification with just the key and assemble the description in the test loop below based on the given key?

Reply:
good idea, done